### PR TITLE
Made initial ES6 changes

### PR DESCRIPTION
**Pull request by spiray**
I made an initial commit to show you what the updated code would look like. Please let me know if this is what you had in mind. This is in response to #28 @lm-n 

Thanks!

### CODE REVIEW

**Comment by MathuraMG** [comment]
@spiray - this looks great! Yes, this is what we were going for, using arrow function, replacing var etc.
I'm curious if you are using a global lint file to detect these changes?

**Comment by MathuraMG** [comment]
Got it!
These changes look good, could you compile them and test with the example code? The steps for this are in the [README](https://github.com/processing/p5.accessibility#development-installation). We will review it right after :) @lm-n

**Comment by spiray** [comment]
This is the one error i got when testing the example. 
image. It looks like that may have been there prior to my changes as i didnt change the name. Does that error mean that the keyword `arguments` shouldn't be used? Can you clone my fork and test?

**Comment by MathuraMG** [comment]
So, it's all working for me - the error that you are seeing is related to [this issue](https://stackoverflow.com/questions/27206974/is-it-ok-to-use-an-argument-named-arguments-in-javascript) in the [getColorName function](https://github.com/processing/p5.accessibility/pull/60/files#diff-0b5bb38c1e06a2f26889d2ea7fc5ab66R26)
So yes, this is an issue to make note of, maybe we can make another issue for this, to be resolved separately. 
Would you have a global lint file on your system stopping this from compiling?
Just to reconfirm, @lm-n , could you check if this works for you? If yes, let's merge this! :)

File: src/registry.js [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
-
-  entityFor: function(name) {
-    for (var i = 0; i < this.registry.length; i++) {
-      var entity = this.registry[i];
-
-      if (entity.handles(name)) {
-        return entity;
-      }
+    registry: [],
+    register: entity => this.registry.push(entity),
```
Comment (lm-n) [inline src/registry.js]: This is giving me an error 'Cannot read property 'push' of undefined'

File: src/baseInterceptor.js [hunk 2]
```diff
@@ -1,252 +1,232 @@
...
+            },
+            this.drawObject = {
+                objectArray: [],
+                objectCount: 0,
+                objectTypeCount: {}
+            },
+            this.isCleared = false;
+    }
+    getColorName(arguments) {
+        if (arguments.length == 4) {
```
Comment (lm-n) [inline src/baseInterceptor.js]: @spiray can you change the variable 'arguments' to 'colorValues'? It should be a quick fix to the error we are getting. Make the change throughout the whole file 😄

**Comment by lm-n** [comment]
@spiray there are some issues in the interceptorFunctions.js files 
`baseInterceptor.call(self);`
due to ES6 "class constructor baseInterceptor cannot be invoked without 'new'" could you take a look at it too?

**Comment by MathuraMG** [comment]
Hmm, I did grunt. But yes, I was a bit confused why I was getting different results, let me check. Thanks for checking @lm-n !

File: src/registry.js (hunk 1 above)
Comment (spiray) [reply src/registry.js]: I changed this back to a standard function for now

File: src/baseInterceptor.js (hunk 2 above)
Comment (spiray) [reply src/baseInterceptor.js]: This is complete

**Comment by spiray** [comment]
@lm-n Regarding the `BaseInterceptor` `class` syntax. Just so i understand, the `textInterceptor` and `gridInterceptor` are inheriting the `BaseInterceptor`? I am just not used to the syntax used here. Once that is confirmed, i will implement ES6 class inheritance and we can make the code much shorter and easier to read!

**Comment by spiray** [comment]
@lm-n @MathuraMG Should I open an issue for testing? It should be relatively easy to write tests for this library. Which testing env would you want to use?

File: src/gridInterceptor/interceptorP5.js [hunk 3]
```diff
@@ -1,68 +1,68 @@
...
+                //concat the new objects and populate the grid
+                //TODO : make this more efficient so that it happens only ONCE per frame count
+                programObjects = gridInterceptor.setupObject.objectArray.concat(gridInterceptor.drawObject.objectArray);
+                gridInterceptor.populateObjectDetails(gridInterceptor.setupObject, gridInterceptor.drawObject, summary, details);
+                gridInterceptor.populateTable(programObjects, document);
+            }
+            if (x.name == 'redraw') { // reset some of the letiables
+                gridInterceptor.drawObject = gridInterceptor.clearletiables(gridInterceptor.drawObject);
+            }
+            return originalFunc.apply(this, arguments);
```
Comment (lm-n) [inline src/gridInterceptor/interceptorP5.js]: Throwing an error here. Try changing the arguments name.

File: src/gridInterceptor/interceptorP5.js [hunk 4]
```diff
@@ -1,68 +1,68 @@
...
-}
+                //concat the new objects and populate the grid
+                //TODO : make this more efficient so that it happens only ONCE per frame count
+                programObjects = gridInterceptor.setupObject.objectArray.concat(gridInterceptor.drawObject.objectArray);
+                gridInterceptor.populateObjectDetails(gridInterceptor.setupObject, gridInterceptor.drawObject, summary, details);
+                gridInterceptor.populateTable(programObjects, document);
+            }
+            if (x.name == 'redraw') { // reset some of the letiables
+                gridInterceptor.drawObject = gridInterceptor.clearletiables(gridInterceptor.drawObject);
+            }
```
Comment (lm-n) [inline src/gridInterceptor/interceptorP5.js]: I suggest changing `clearletiables` to `clearLetiables`

File: src/gridInterceptor/interceptorP5.js [hunk 5]
```diff
@@ -1,68 +1,68 @@
...
-        gridInterceptor.populateTable(details, gridInterceptor.setupObject);
-      } else if (frameCount == 1 || frameCount % 20 == 0) {
-        gridInterceptor.drawObject =
-        gridInterceptor.populateObject(x, arguments, gridInterceptor.drawObject, details, true);
-        gridInterceptor.isCleared = false;
+            if (frameCount == 0) { // for setup
+                table.innerHTML = '';
+                details.innerHTML = '';
+                summary.innerHTML = '';
+                gridInterceptor.createShadowDOMElement(document);
```
Comment (lm-n) [inline src/gridInterceptor/interceptorP5.js]: throwing an error here, might be fixed when line 65 problem is fixed

File: src/gridInterceptor/interceptorP5.js (hunk 3 above)
Comment (spiray) [reply src/gridInterceptor/interceptorP5.js]: @lm-n I don't think this a variable that we can just change. This is actually using the arguments array. This should not throw an error. Can you show me the error that you are getting?

File: src/gridInterceptor/interceptorP5.js (hunk 4 above)
Comment (spiray) [reply src/gridInterceptor/interceptorP5.js]: Done, That was done in error from a global replace.

File: src/gridInterceptor/interceptorP5.js (hunk 5 above)
Comment (spiray) [reply src/gridInterceptor/interceptorP5.js]: This should be good to go.

**Comment by spiray** [comment]
I worked on `_baseEntity.js` and i think i got it right but the way the `arguments` word is used and the way the arguments are passed is confusing as to the scope of each variable. If i can do a session with either @lm-n or @MathuraMG to clarify the code i can make much better progress. I am mostly doing this without much knowledge of the code structure or what it really does.

**Comment by MathuraMG** [comment]
@spiray - that sounds like a good idea. Would you want to talk about it sometime thsi week? In the meantime, we will add a write-up to the readme explaining more about what the code does.

**Comment by spiray** [comment]
@MathuraMG Yes, i can discuss it this week. My email is spira.joe214@gmail.com so you can reach me. Thanks!

**Comment by spiray** [comment]
I am closing this pr as i started fresh in #68.

**Pull request closed without merging**